### PR TITLE
junit for Profile class

### LINKED ISSUE #124: add Junit test for Profile class

### PR DESCRIPTION
**Pull request by froststein**
Fixes #124

### CODE REVIEW

**Review by warrencxw — APPROVED** [review]
Looks good to merge.

File: src/test/java/seedu/splitlah/data/ProfileTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        try {
+            Session retrievedSession = manager.getProfile().getSession(sessionIdToTest);
+            assertEquals(sessionIdToTest, retrievedSession.getSessionId());
+        } catch (InvalidDataException invalidDataException) {
+            fail();
+        }
+    }
+
+    /**
+     * Checks if an exception is properly thrown when a session unique identifier is not in the list of sessions.
```
Comment (ivanaitzliddat) [inline src/test/java/seedu/splitlah/data/ProfileTest.java:87]: Possible to change this to "Checks if an exception is properly thrown when a Session object with a specified session unique identifier is not found."

File: src/test/java/seedu/splitlah/data/ProfileTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    @Test
+    public void hasSessionId_inputContainsNonExistingSessionId_false() {
+        int sessionIdToTest = 10;
+        boolean isExists = manager.getProfile().hasSessionId(sessionIdToTest);
+        assertEquals(false, isExists);
+    }
+
+
+    /**
+     * Checks if the session is returned when a session unique identifier is in the list of sessions.
```
Comment (ivanaitzliddat) [inline src/test/java/seedu/splitlah/data/ProfileTest.java:73]: Possible to change this to "Checks if an exception is properly thrown when a Session object with a specified session unique identifier is found."

File: src/test/java/seedu/splitlah/data/ProfileTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+     */
+    @Test
+    public void hasSessionId_inputContainsExistingSessionId_true() {
+        int sessionIdToTest = 1;
+        boolean isExists = manager.getProfile().hasSessionId(sessionIdToTest);
+        assertEquals(true, isExists);
+    }
+
+    /**
+     * Checks if method returns false when a session unique identifier does not exist within the list of session.
```
Comment (ivanaitzliddat) [inline src/test/java/seedu/splitlah/data/ProfileTest.java:62]: Possible to change this to "Checks if method returns false when a Session object with a specified session unique identifier is not found."

File: src/test/java/seedu/splitlah/data/ProfileTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+     */
+    @Test
+    public void hasSessionName_inputContainsNonExistingSessionName_false() {
+        String sessionNameToTest = "School gathering";
+        boolean isExists = manager.getProfile().hasSessionName(sessionNameToTest);
+        assertEquals(false, isExists);
+    }
+
+    /**
+     * Checks if method returns true when a session unique identifier exists within the list of session.
```
Comment (ivanaitzliddat) [inline src/test/java/seedu/splitlah/data/ProfileTest.java:52]: Possible to change this to "Checks if method returns true when a Session object with a specified session unique identifier is found."

File: src/test/java/seedu/splitlah/data/ProfileTest.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+     */
+    @Test
+    public void hasSessionName_inputContainsExistingSessionName_true() {
+        String sessionNameToTest = "Class outing";
+        boolean isExists = manager.getProfile().hasSessionName(sessionNameToTest);
+        assertEquals(true, isExists);
+    }
+
+    /**
+     * Checks if method returns false when a session name does not exist within the list of session.
```
Comment (ivanaitzliddat) [inline src/test/java/seedu/splitlah/data/ProfileTest.java:42]: Possible to change this to "Checks if method returns false when a Session object with a specified session name is not found."

File: src/test/java/seedu/splitlah/data/ProfileTest.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        String sessionOneArgs = "session /create /n Class outing /d 15-02-2022 /pl Alice Bob";
+        String sessionTwoArgs = "session /create /n Family gathering /d 20-02-2022  /pl Eves Mallory";
+        Command createSessionOne = Parser.getCommand(sessionOneArgs);
+        createSessionOne.run(manager);
+        Command createSessionTwo = Parser.getCommand(sessionTwoArgs);
+        createSessionTwo.run(manager);
+    }
+
+    /**
+     * Checks if method returns true when a session name exists within the list of session.
```
Comment (ivanaitzliddat) [inline src/test/java/seedu/splitlah/data/ProfileTest.java:32]: Possible to change this to "Checks if method returns true when a Session object with a specified session name is found."

**Review by ivanaitzliddat — CHANGES_REQUESTED** [review]
Looks good overall, but possible to change the Javadoc as per the comments.

**Review by ivanaitzliddat — APPROVED** [review]
Looks good to merge

**Pull request merged**